test/FixedPoint.spec.ts

```diff
@@ -91,6 +80,80 @@ describe('FixedPoint', () => {
         BigNumber.from(2).pow(224)
       )
     })
+    it('max without overflow, largest fixed point', async () => {
+      const maxMultiplier = BigNumber.from('4294967296')
+      expect((await fixedPoint.mul([BigNumber.from(2).pow(224).sub(1)], maxMultiplier))[0]).to.eq(
+        BigNumber.from('115792089237316195423570985008687907853269984665640564039457584007908834672640')
+      )
+      await expect(fixedPoint.mul([BigNumber.from(2).pow(224).sub(1)], maxMultiplier.add(1))).to.be.revertedWith(
+        'FixedPoint: MULTIPLICATION_OVERFLOW'
+      )
+    })
+    it('max without overflow, smallest fixed point', async () => {
+      const maxUint = BigNumber.from(2).pow(256).sub(1)
+      expect((await fixedPoint.mul([BigNumber.from(1)], maxUint))[0]).to.eq(maxUint)
+      await expect(fixedPoint.mul([BigNumber.from(2)], maxUint)).to.be.revertedWith(
+        'FixedPoint: MULTIPLICATION_OVERFLOW'
+      )
+    })
+  })
+
+  describe('#muli', () => {
+    it('works for 0', async () => {
+      expect(await fixedPoint.muli([BigNumber.from(0).mul(Q112)], BigNumber.from(1))).to.eq(BigNumber.from(0))
+      expect(await fixedPoint.muli([BigNumber.from(1).mul(Q112)], BigNumber.from(0))).to.eq(BigNumber.from(0))
+    })
+
+    it('works for 3*2', async () => {
+      expect(await fixedPoint.muli([BigNumber.from(3).mul(Q112)], BigNumber.from(2))).to.eq(BigNumber.from(6))
+    })
+
+    it('works for 3*-2', async () => {
+      expect(await fixedPoint.muli([BigNumber.from(3).mul(Q112)], BigNumber.from(-2))).to.eq(BigNumber.from(-6))
+    })
+
+    it('overflow', async () => {
+      await expect(fixedPoint.muli([BigNumber.from(1).mul(Q112)], BigNumber.from(2).pow(144))).to.be.revertedWith(
+        'FixedPoint: MULTIPLICATION_OVERFLOW'
+      )
+      await expect(
+        fixedPoint.muli([BigNumber.from(1).mul(Q112)], BigNumber.from(2).pow(144).mul(-1))
+      ).to.be.revertedWith('FixedPoint: MULTIPLICATION_OVERFLOW')
+    })
+    it('max without overflow, largest fixed point', async () => {
+      const maxMultiplier = BigNumber.from('4294967296')
```

Comment:
i know this is 2^32 but it would be good to derive it so it's more obvious (BigNumber.from(2).pow(32)) 

(above as well)